meteor/server/api/rest/api.ts

```diff
@@ -0,0 +1,1494 @@
+		logger.info(`koa PUT: resetPlaylist ${rundownPlaylistId}`)
+
+		check(rundownPlaylistId, String)
+		return await serverAPI.resetPlaylist(connection, event, rundownPlaylistId)
+	}
+)
+
+sofieAPIRequest<{ playlistId: string }, { partId: string }, void>(
+	'put',
+	'/playlists/:playlistId/setNextPart',
+	new Map([
+		[404, UserErrorMessage.RundownPlaylistNotFound],
+		[412, UserErrorMessage.PartNotFound],
+	]),
+	async (serverAPI, connection, event, params, body) => {
+		const rundownPlaylistId = protectString<RundownPlaylistId>(params.playlistId)
+		const partId = protectString<PartId>(body.partId)
+		logger.info(`koa PUT: setNextPart ${rundownPlaylistId} ${partId}`)
+
+		check(rundownPlaylistId, String)
+		check(partId, String)
+		return await serverAPI.setNextPart(connection, event, rundownPlaylistId, partId)
+	}
+)
+
+sofieAPIRequest<{ playlistId: string }, { segmentId: string }, void>(
+	'put',
+	'/playlists/:playlistId/setNextSegment',
+	new Map([
+		[404, UserErrorMessage.RundownPlaylistNotFound],
+		[412, UserErrorMessage.PartNotFound],
+	]),
+	async (serverAPI, connection, event, params, body) => {
+		const rundownPlaylistId = protectString<RundownPlaylistId>(params.playlistId)
+		const segmentId = protectString<SegmentId>(body.segmentId)
+		logger.info(`koa PUT: setNextSegment ${rundownPlaylistId} ${segmentId}`)
+
+		check(rundownPlaylistId, String)
+		check(segmentId, String)
+		return await serverAPI.setNextSegment(connection, event, rundownPlaylistId, segmentId)
+	}
+)
+
+sofieAPIRequest<{ playlistId: string }, { fromPartInstanceId?: string }, void>(
+	'put',
+	'/playlists/:playlistId/take',
+	new Map([
+		[404, UserErrorMessage.RundownPlaylistNotFound],
+		[412, UserErrorMessage.TakeNoNextPart],
+	]),
+	async (serverAPI, connection, event, params, body) => {
+		const rundownPlaylistId = protectString<RundownPlaylistId>(params.playlistId)
+		const fromPartInstanceId = body.fromPartInstanceId
+		logger.info(`koa PUT: take ${rundownPlaylistId}`)
+
+		check(rundownPlaylistId, String)
+		check(fromPartInstanceId, Match.Optional(String))
+		return await serverAPI.take(connection, event, rundownPlaylistId, protectString(fromPartInstanceId))
+	}
+)
+
+sofieAPIRequest<{ playlistId: string; sourceLayerId: string }, never, void>(
+	'delete',
+	'/playlists/:playlistId/sourceLayer/:sourceLayerId',
+	new Map([
+		[404, UserErrorMessage.RundownPlaylistNotFound],
+		[412, UserErrorMessage.InactiveRundown],
+	]),
+	async (serverAPI, connection, event, params, _) => {
+		const playlistId = protectString<RundownPlaylistId>(params.playlistId)
+		const sourceLayerId = params.sourceLayerId
+		logger.info(`koa DELETE: sourceLayer ${playlistId} ${sourceLayerId}`)
+
+		check(playlistId, String)
+		check(sourceLayerId, String)
+		return await serverAPI.clearSourceLayer(connection, event, playlistId, sourceLayerId)
+	}
+)
+
+sofieAPIRequest<{ playlistId: string; sourceLayerId: string }, never, void>(
+	'post',
```

Comment:
Why is this a `post` when _take_ and _executeAdLib_ is a `put`? All of these perform the same _sort_ of action, I think.

Reply:
This has been updated to `put`